src/clean_air/data/extract_metadata.py

```diff
@@ -106,11 +109,21 @@ def extract_metadata(
                 temporal_extent = TemporalExtent(intervals=time_interval)
             else:
                 temporal_extent = TemporalExtent(values=time_list)
-            total_temporal_extent_list.update(time_list)
-
-        if len(cube.coords(axis='z')) == 1:
-            vertical_extent = VerticalExtent(cube.coord(axis='z').points)
-            total_vertical_extent_list.append(cube.coord(axis='z').points)
+            total_temporal_extent_set.update(time_list)
+
+        #TODO: Currently parameter vert ext is array of all non-NaN values,
+        # total vert extent is list of unique values.
+
+        # list of coords that may represent z axis, in ascending order of preference
+        z_options = [cube.coords('altitude'), cube.coords(axis='z')]
+        for item in z_options:
+            if len(item) == 1:
+                z_coord = item[0].points
+        if z_coord is not None:
+            if np.ma.isMaskedArray(z_coord):
+                z_coord = z_coord.compressed()
+            vertical_extent = VerticalExtent(z_coord)
+            total_vertical_extent_set.update(z_coord)
```

Comment:
Have to admit, I don't fully understand some of this (not familiar with `coord.compressed`, and I don't quite understand the looping here) but I trust your judgement.

Reply:
This part just checks if our z coordinate data is a masked array, and if it is, uses this [function](https://numpy.org/doc/stable/reference/generated/numpy.ma.MaskedArray.compressed.html) to get a ndarray of our data that will then work with our `VerticalExtent` object!

Reply:
Cool.  I like that!